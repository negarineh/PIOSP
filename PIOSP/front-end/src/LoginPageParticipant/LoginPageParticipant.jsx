import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { userActions } from '../_actions';
import '../components/stylesheets/Login.css';

// this component using jwt in action function and 
// in action we check the json web token make sure participant is authenticated to atthend the survey
class LoginPageParticipant extends React.Component {

    componentDidMount() {
        this.props.dispatch(userActions.attend());
    }
    
    render() {
        return (
            <div >
            </div>
        );
    }
}

function mapStateToProps(state) {
    const { loggingIn } = state.authentication;
    return {
        loggingIn
    };
}

const connectedLoginPage = connect(mapStateToProps)(LoginPageParticipant);
export { connectedLoginPage as LoginPageParticipant }; 

LoginPageParticipant.defaultProps = {
    dispatch: () => {}
}

LoginPageParticipant.proptypes = {
    dispatch: PropTypes.func.isRequired
}