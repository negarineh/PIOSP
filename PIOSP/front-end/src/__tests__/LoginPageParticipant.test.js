import React from "react";
import Enzyme, { shallow } from "enzyme";
import assert from "assert";
import {LoginPageParticipant} from "../LoginPageParticipant";
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import { store } from '../_helpers';
import { Provider } from 'react-redux';
 
Enzyme.configure({ adapter: new Adapter() });

// unit tests for the UserDeletePrompt component
describe('Login Page Participant component', () => {
  describe('render()', () => {
    it('should render the component', () => {
      const props = {dispatch: () => {}};
      const wrapper = shallow(<Provider store={store}><LoginPageParticipant {...props}/></Provider>);
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
      assert.equal(wrapper.length, 1);
    });
  });
});
