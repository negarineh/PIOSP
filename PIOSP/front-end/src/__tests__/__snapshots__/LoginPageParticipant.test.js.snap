// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page Participant component render() should render the component 1`] = `
<Connect(LoginPageParticipant)
  dispatch={[Function]}
/>
`;
